course-clusters/clusters.py

```diff
@@ -0,0 +1,116 @@
+import string
+from gensim.models.doc2vec import Doc2Vec, TaggedDocument
+from load_course_info import loadCourseDesc, getSubjects
+
+subjects_url = "https://classes.cornell.edu/api/2.0/config/subjects.json?roster=FA23"
+subjects = getSubjects(subjects_url)
+
+urls = [f"https://classes.cornell.edu/api/2.0/search/classes.json?roster=FA23&subject={sub}" for sub in subjects]
+
+def fetchDescriptions(urls):
+    descriptions = []
+    for x in urls:
+        descriptions += loadCourseDesc(x)
+    print(len(descriptions))
+    return descriptions
+
+
+course_descriptions = fetchDescriptions(urls)
+def preprocess_text(text):
+    """ Tokenization and preprocessing with custom stopwords"""
+    custom_stopwords = ["of", "the", "in", "and", "on", "an", "a", "to"]
+    strong_words = ["technology","calculus","business", "Artificial Intelligence", "First-Year Writing", "computer","python","java","economics","US","writing","biology","chemistry", "physics", "engineering","ancient"
+                    "programming", "algorithms", "data structures","art","software","anthropology" "databases","fiction","mathematics", "history","civilization"]
+
+    translator = str.maketrans("", "", string.punctuation)
+    text = text.lower()
+    text = text.translate(translator)
+    tokens = text.split()
+    tokens = [token for token in tokens if token not in custom_stopwords]
+    for word in strong_words:
+        if word in tokens:
+            tokens += [word] * 10 
+
+    return " ".join(tokens)
+
+def removeStopwords():
+    "Removes stopwords for all the descriptions "
+    preprocessed = []
+    for desc in course_descriptions:
+        if desc:
+            preprocessed.append(preprocess_text(desc))
+    return preprocessed
+
+preprocessed_descriptions = removeStopwords()
```

Comment:
Let's also move this preprocessing out. Separate the preprocessing/web scrapping with the actual model itself. Can we also create some persistent objects to store the web scrapping data and the preprocessing so we don't always have to do that?